### ISSUE TITLE
Improvement Request to a modern and isolate webui and a server most modular

### ISSUE THREAD

**Issue by rafaelvieiras** [DEFENDORe/pseudotv#19]
First, I'm very happy to see other person with a project like this. I old user PseudoTv on XBMC and after PseudoTv Live on Kodi.
I'm collaborated some times with PseudoTV Live and many times with PseudoTv (classic), but I don't use Kodi more. 

My dream is to watch my PseudoTv configuration on a simple iptv client, without exclusive tool requirements like "use kodi only", "use plex only".

I'm tried many, and many times to start to develop a "PseudoPVR" service, but, I never has time to do, and always I try to do this on bad way possible. Seeing your code, I'm loving the way you do, a just re-streaming and using external services to get medias information.

I would like to suggest a new perspective for the web user interface.

Currently, the ‘user interface’ was written in Angular 1.7.9. Any improvement or development, including performance,
it's hard to do.
So, my first suggestion is to migrate the webui to Angular 9, with a structured project and served within the webui folder.

Another point concerns the format of how the Plex configuration is embedded within the client and in hard-code on the server. The problem is that it does not make it possible or simple to implement new service ‘interfaces’, like Jellyfin.
So, my suggestion is:
- [Server] creation of a "Driver" that will be used to communicate with the chosen service, be it Plex, Jellyfin or Emby.
- [Client] remove the logic related to the Plex service on the client, and leave the management to the Server. Basically the Client will ask the Server for information, without knowing what logic was used to get this information.
For example, this code on client:
```js
        getLibrary: async (server) => {
            var client = new Plex(server)
            const res = await client.Get('/library/sections')
...
```
Can be:
```js
        getLibrary: async () => {
            const res = await ServerService.GetLibary('plex') // Make request to localhost:8000/library?service=plex
...
```
With that, the server decides which library it will call, and user Driver class to do this. With this, may services can used, and on future, we can have a youtube plugin, to create channel with Youtube videos, o use commercial and bumbles form youtube. And the best point is, with  more services supported, more collaboration on development.

I would like to contribute to the project, but I was an Emby user and now I am a Jellyfin user, but I have no interest in migrating to plex.

I could start working on those better ones.

**Comment by powdermilkman** [comment]
I agree that we need to move to a more modern version of angular. If you want to get that started we would welcome the pr. And separation of client and server is also something we need to move toward, no reason to keep building this with only plex in mind.
